snuba/subscriptions/worker.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import NamedTuple, Mapping, MutableSequence, Optional, Sequence
+
+from snuba.subscriptions.consumer import Tick
+from snuba.subscriptions.data import Subscription
+from snuba.subscriptions.executor import SubscriptionExecutor
+from snuba.subscriptions.scheduler import ScheduledTask, Scheduler
+from snuba.utils.streams.batching import AbstractBatchWorker
+from snuba.utils.streams.producer import Producer
+from snuba.utils.streams.types import Message, Topic
+from snuba.web.query import ClickhouseQueryResult
+
+
+class SubscriptionResult(NamedTuple):
+    task: ScheduledTask[Subscription]
+    result: ClickhouseQueryResult
+
+
+class SubscriptionWorker(AbstractBatchWorker[Tick, Sequence[SubscriptionResult]]):
+    def __init__(
+        self,
+        executor: SubscriptionExecutor,
+        schedulers: Mapping[int, Scheduler[Subscription]],
+        producer: Producer[SubscriptionResult],
+        topic: Topic,
+    ) -> None:
+        self.__executor = executor
+        self.__schedulers = schedulers
+        self.__producer = producer
+        self.__topic = topic
+
+    def process_message(
+        self, message: Message[Tick]
+    ) -> Optional[Sequence[SubscriptionResult]]:
+        results: MutableSequence[SubscriptionResult] = []
+
+        tick = message.payload
+        for task in self.__schedulers[message.partition.index].find(tick.timestamps):
+            results.append(
+                SubscriptionResult(task, self.__executor.execute(task, tick).result())
+            )
+
+        return results
+
+    def flush_batch(self, batch: Sequence[Sequence[SubscriptionResult]]) -> None:
+        for results in batch:
+            for result in results:
+                self.__producer.produce(self.__topic, result).result()
```

Comment:
At some point, we'll probably want to be able to write to different topics dynamically based on the subscription. I don't know what this looks like, but I'd imagine if we have metric alert subscriptions and data science subscriptions all in the same topic it's going to get noisy and bloated.

I don't think one subscription per topic is a good idea. I also don't think users should be explicitly providing the destination topic, either. Something in between probably makes sense. (If we had users, that'd probably be a good way to segment these, but that's a whole different can of worms.)

Reply:
It might be as simple as just having a subscription type. That might be built into the key, the worker might just decide the topic based on that, or maybe we have a `TopicPartitioner` that decides where it goes.